docs/design/proposals/status-conditions.md

```diff
@@ -198,18 +198,84 @@ Things to note:
 
 ##### Resource Graph Instance
 
-This one is currently pretty tricky because it accepts conditions to be defined
-inside the RGD Instance. But at minimum and by default the controller adds:
+The Resource Graph Instance currently uses a single condition:
 
 - InstanceSynced
 
-To be able to use a condition manager, we would need a signal from the provided
-status shape to understand if the defined condition is a dependent condition or
-just informational.
+Based on analysis of the instance reconciler, the major sections of work can be
+broken down into the following sub-conditions that align with the reconciliation
+phases:
+
+```
+Ready
+├─ InstanceManaged - Instance finalizers and labels are properly set
+├─ GraphResolved - Runtime graph has been created and resources resolved
+└─ ResourcesReady - All resources in the graph are created and ready
+```
+
+The reconciler performs the following major work sections:
+
+1. **Instance Setup**: Setting managed state and finalizers (InstanceManaged)
+2. **Runtime Creation**: Creating the ResourceGraphDefinition runtime and resolving resources (GraphResolved)
+3. **Resource Reconciliation**: Processing each resource in topological order - creation, updates, and readiness checks (ResourcesReady)
+4. **Instance Deletion**: Deleting resources in reverse topological order
 
 A similar example applies to Resource Graph Instance as shown under Resource
 Graph Definition.
 
+`Current Instance Status`
+
+```
+status:
+  state: ACTIVE
+  conditions:
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: Instance reconciled successfully
+    reason: ReconciliationSucceeded
+    status: "True"
+    type: InstanceSynced
+```
+
+`Proposed Update`
+
+```
+status:
+  state: ACTIVE
+  conditions:
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: ""
+    observedGeneration: 1
+    reason: "AllReady"
+    status: "True"
+    type: Ready
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: Instance is properly managed with finalizers and labels
+    observedGeneration: 1
+    reason: "Managed"
+    status: "True"
+    type: InstanceManaged
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: Runtime graph created and all resources resolved
+    observedGeneration: 1
+    reason: "Resolved"
+    status: "True"
+    type: GraphResolved
+  - lastTransitionTime: "2025-08-07T21:31:02Z"
+    message: All resources are created and ready
+    observedGeneration: 1
+    reason: "AllResourcesReady"
+    status: "True"
+    type: ResourcesReady
```

Comment:
Do we want to also add a condition per managed resource? i'm thinking it might be noisy but helpful to debug what's wrong and at which step things are failing.

Reply:
I really would like to, maybe it could be an option? I think it would help debug and understand but also it leaks a lot of details that you might not want to leak. For example, if a resource makes 100 children, not sure if we want to show 100 sub-conditions :D 

What do others think? We could start with one and do an experiment where we do verbose mode via an annotation and if that is interesting we could roll it into the real spec settings on RGD?

Reply:
If at all we want to add resource level or coffman-graham-level subcondition it should be for cases where we saw a failure in the resource. During normal operation we dont want them in subconditions. Also beware of the object size limitation in etcd for k8s. We dont want status updates to fail because we exceeded object size. 

TBH we are too early for this pattern. Lets follow up in a separate bug/feature req for this.